Plugins/ExaTrkX/src/OnnxMetricLearning.cpp

```diff
@@ -57,7 +57,7 @@ void OnnxMetricLearning::buildEdgesWrapper(std::vector<float>& embedFeatures,
 }
 
 std::tuple<std::any, std::any> OnnxMetricLearning::operator()(
-    std::vector<float>& inputValues, std::size_t, int) {
+    std::vector<float>& inputValues, std::size_t, torch::Device) {
```

Comment:
I think this will cause compilation issue if ACTS is anly built with Onnx and not pytorch (using `ACTS_EXATRKX_ENABLE_ONNX` but not `ACTS_EXATRKX_ENABLE_TORCH`). So I guess you need to find a way arround that before this can be merged

Reply:
Hello @Corentin-Allaire thanks for the comment. Actually `ACTS_BUILD_PLUGIN_EXATRKX` requires `Torch`(see [CMake](https://github.com/acts-project/acts/blob/main/CMakeLists.txt#L310C16-L310C21)) anyways so `ONNX` can also find torch(and `ACTS_EXATRKX_ENABLE_TORCH` is looking for `Torch Scatter`).  Another example is that `OnnxMetricLearning::buildEdgesWrapper` also use `torch::Device`. 

And actually I don't find a way to specify a GPU for ONNX to run on. In this case would it be easier to leave things unchanged for ONNX part? 
I then further test the compilation with ONNX but it failed. I found both ONNX and Torch inherited from the same base class. So I changed the base class which means the implementation of additional functions on the ONNX side is required. But given no way to specify the GPU, it's not necessary.  I'm thinking just to restore the base class and add more features ONLY to the Torch side, what do you think of this approach? 

@xju2 Also tagging Xiangyang for advice. 

FYI last related commit: 
https://github.com/acts-project/acts/pull/2553

Reply:
I missed the fact that `ACTS_BUILD_PLUGIN_EXATRKX` requires `Torch`. In that case you can keep it as is it is not an issue.
I am not sure I understand your second paragraph, do you mean that your current implementation doesn't work with onnx ? In that case yes I would use torch::Device on one side and int on the other

Reply:
the second graph is about the OnnxEdgeClassifier and OnnxMetricLearning  compilation error because of the difference with the base function. 

I think currently the ONNX execution of the ExaTrkX model is not running on GPU(`Ort::SessionOptions` needs `AppendExecutionProvider_CUDA`, which is absent in the code) so I just add the default CPU device to accommodate the  [OnnxEdgeClassifier.hpp](https://github.com/hrzhao76/acts/commit/b6e52ef8c12911e13b1761539b9999e179b3a92e#diff-81367aae9c9ea98035b86c44620d5c168929d4cbfb156732432601f2d708f3d8) and [OnnxMetricLearning.hpp](https://github.com/hrzhao76/acts/commit/b6e52ef8c12911e13b1761539b9999e179b3a92e#diff-6ea9a85f34284934fe47f2d6d64d1284fe6d3e0e89619bb774caa92427d9413e) to the base class. 

 Let me know what you think.